packages/openapi-generator/src/parser/api.ts

```diff
@@ -0,0 +1,94 @@
+import { $Refs } from '@apidevtools/swagger-parser';
+import { flat, pascalCase } from '@sap-cloud-sdk/util';
+import { OpenAPIV3 } from 'openapi-types';
+import { methods, OpenApiApi } from '../openapi-types';
+import { apiNameExtension, defaultApiName } from '../extensions';
+import { parseOperation } from './operation';
+import { OperationInfo } from './operation-info';
+import { ensureUniqueOperationIds, nameOperations } from './operation-naming';
+
+/**
+ * Collect and parse all APIs of an `OpenAPIV3.Document`.
+ * @param document The OpenApi document to parse.
+ * @param refs List of crossreferences that can occur in the document.
+ * @returns A flat list of parsed APIs.
+ */
+export function parseApis(
+  document: OpenAPIV3.Document,
+  refs: $Refs
+): OpenApiApi[] {
+  const operationsByApis = getOperationsByApis(document);
+
+  return Object.entries(operationsByApis).map(([name, operations]) => ({
```

Comment:
could you add the type of `operations`? I think it is already `OperationsInfo`.

Reply:
done